internal/ingress/controller/store/store.go

```diff
@@ -358,13 +367,27 @@ func New(checkOCSP bool,
 			validOld := class.IsValid(oldIng)
 			validCur := class.IsValid(curIng)
 			if !validOld && validCur {
+				if curIng.Spec.Backend != nil && disableCatchAll {
+					klog.Infof("ignoring update for catch-all ingress %v/%v because of --disable-catch-all", curIng.Namespace, curIng.Name)
+					return
+				}
+
 				klog.Infof("creating ingress %v based on annotation %v", curIng.Name, class.IngressKey)
 				recorder.Eventf(curIng, corev1.EventTypeNormal, "CREATE", fmt.Sprintf("Ingress %s/%s", curIng.Namespace, curIng.Name))
 			} else if validOld && !validCur {
 				klog.Infof("removing ingress %v based on annotation %v", curIng.Name, class.IngressKey)
+				// FIXME: this does not actually delete the Ingress resource.
+				// The existing one will be updated with latest changes and invalid ingress.class will be missed.
 				recorder.Eventf(curIng, corev1.EventTypeNormal, "DELETE", fmt.Sprintf("Ingress %s/%s", curIng.Namespace, curIng.Name))
 			} else if validCur && !reflect.DeepEqual(old, cur) {
-				recorder.Eventf(curIng, corev1.EventTypeNormal, "UPDATE", fmt.Sprintf("Ingress %s/%s", curIng.Namespace, curIng.Name))
+				if curIng.Spec.Backend != nil && disableCatchAll {
+					klog.Infof("ignoring update for catch-all ingress %v/%v because of --disable-catch-all", curIng.Namespace, curIng.Name)
+					// FIXME: this does not actually delete the Ingress resource.
+					// The existing one will be updated with latest changes.
+					recorder.Eventf(curIng, corev1.EventTypeNormal, "DELETE", fmt.Sprintf("Ingress %s/%s", curIng.Namespace, curIng.Name))
```

Comment:
Why are you not deleting ingress here? To keep it consistent with the `valid class` behaviour? If so then why are you recording `DELETE` event rather than `UPDATE`?

Reply:
This has the same behavior has the `valid class` check. It records a `DELETE` event, but don't actually delete it.

I wanted more feedback on this issue before trying any fix here.
I've quickly tested to call the DeleteFunc handler when needed here and it works as expected. But I'm not 100% sure this is the solution.

Also, I think this should be fixed in a 2nd PR (that can be merge before this one).

Reply:
👍